reconcile/openshift_groups.py

```diff
@@ -58,32 +61,59 @@
 """
 
 
-def fetch_current_state():
-    gqlapi = gql.get_api()
-    clusters = gqlapi.query(CLUSTERS_QUERY)['clusters']
-    current_state = []
-    oc_map = {}
-
+def get_cluster_state(group_items, oc_map):
+    results = []
+    cluster_info = group_items["cluster_info"]
+    cluster = cluster_info['name']
+    oc = oc_map[cluster]
+    group_name = group_items["group_name"]
+    group = oc.get_group_if_exists(group_name)
+    if group is None:
+        return results
+    for user in group['users'] or []:
+        results.append({
+            "cluster": cluster,
+            "group": group_name,
+            "user": user
+        })
+    return results
+
+
+def create_groups_list(clusters):
+    groups_list = []
     for cluster_info in clusters:
         groups = cluster_info['managedGroups']
         if groups is None:
             continue
+        for group_name in groups:
+            groups_list.append({
+                    "cluster_info": cluster_info,
+                    "group_name": group_name
+                })
+    return groups_list
+
 
+def create_oc_map(clusters):
+    oc_map = {}
+    for cluster_info in clusters:
         cluster = cluster_info['name']
         oc = openshift_resources.obtain_oc_client(oc_map, cluster_info)
         oc_map[cluster] = oc
+    return oc_map
 
-        for group_name in groups:
-            group = oc.get_group_if_exists(group_name)
-            if group is None:
-                continue
-            for user in group['users'] or []:
-                current_state.append({
-                    "cluster": cluster,
-                    "group": group_name,
-                    "user": user
-                })
 
+def fetch_current_state():
+    gqlapi = gql.get_api()
+    clusters = gqlapi.query(CLUSTERS_QUERY)['clusters']
+    current_state = []
+    oc_map = create_oc_map(clusters)
+
+    pool = ThreadPool(10)
```

Comment:
can we have the `10` be passed in as a variable like in other `threaded` (hint, hint) integrations?